packages/salesforcedx-vscode-soql/src/commonUtils.ts

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { TextDocument } from 'vscode';
+
+export function getDocumentName(document: TextDocument): string {
+  const documentPath = document.uri.fsPath;
+  return documentPath.split('/').pop() || '';
+}
```

Comment:
Do you support untitled document?

Reply:
@xyc it has to be a .soql file extension for SOQL VS Code extension to activate.

Reply:
@praksb what i mean for untitled document is you can do Command Palette -> Change Language Mode -> soql

<img width="758" alt="Screen Shot 2020-09-23 at 2 04 22 PM" src="https://user-images.githubusercontent.com/679275/94069594-b8cb8c00-fda5-11ea-9fd9-04989d96a290.png">

Reply:
activation event has `onLanguage:soql` so I think it would activate in this case

Reply:
That would be a good test case to try.

Reply:
Yes the extension activates, if not already active, if you change the language mode to soql. However, the customer can not open the soql builder unless the file extension is `.soql`.
<img width="1625" alt="untitled file" src="https://user-images.githubusercontent.com/25188632/94074792-38158b80-fdb7-11ea-8da8-514e70e30189.png">

If however I have a `.soql` file ext but I change the language association to HTML, then I can still open the Query Builder UI.
<img width="1625" alt="html file" src="https://user-images.githubusercontent.com/25188632/94074923-6eeba180-fdb7-11ea-939a-4b889dc6333c.png">

Lastly, if I have actual HTML in the file with the `.soql` ext and open the UI then build a query, it will write over the contents of the file and WAD.

So bottom line I don't think there is anything handle differently there. If the user goes out of their way to mess things up and overwrite their file I'm not sure thats on us.

@xyc Are you thinking of a different scenario?

Reply:
I looked up custom editor API and it doesn't seem to support untitled doc scenario, so I'll take that soql builder is not available in untitled docs for now. I'm curious if you could still run a query in untitled docs and _utilize the query results webview_, using "Execute SOQL Query with Selected Text".

I initially had some concerns about `document.uri.fsPath`, it turned out to be just `Untitled` in untitled documents so it won't be an issue here.

Reply:
>I'm curious if you could still run a query in untitled docs and utilize the query results webview, using "Execute SOQL Query with Selected Text".

I don't see why not, we will probably have to make a few changes to the expose the right stuff for the webviewService to be consumed by other extensions or commands, but ultimately the service just needs the records from an API response to render the table.